### PR TITLE
Add skinning clarifications

### LINKED ISSUE #1383: [Clarification] Should I animate nodes that are NOT part of the scene?
Hey,

If one of the nodes that is being animated is not directly part of the scene (for example, it may be a node from a skeleton hierarchy that is not contained within the scene itself), should I animate it or not?

For an example, see the attached gltf file. If I load it with ThreeJs, it does not animate, because the skeleton nodes are not part of the scene. But this is not specified in the specs anywhere as far as I can see. What should I do? Animate them or not?

[SimpleSkinNoChildren.txt](https://github.com/KhronosGroup/glTF/files/2175287/SimpleSkinNoChildren.txt)

### PR DESCRIPTION
**Pull request by lexaknyazev**
Fixes #1383 and #1270. 
Before merging, "Property Reference" section of the spec needs to be updated as well.

### CODE REVIEW

File: specification/2.0/schema/skin.schema.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
         },
         "skeleton": {
             "allOf": [ { "$ref": "glTFid.schema.json" } ],
-            "description": "The index of the node used as a skeleton root. When undefined, joints transforms resolve to scene root."
+            "description": "The index of the node used as a skeleton root.",
+            "gltf_detailedDescription": "The index of the node used as a skeleton root. The node must be the closest common root of the joints hierarchy or a direct or indirect parent node of the closest common root."
```
Comment (donmccurdy) [inline specification/2.0/schema/skin.schema.json]: This no longer specifies what happens when the `skeleton` property is omitted... should we say that the joints must have a common root in a scene node? That may be a breaking change compared to `resolves to scene root` from before. Or should we say that implementations may try to find a common root, which may not exist, or may use the scene root?

Comment (lexaknyazev) [reply specification/2.0/schema/skin.schema.json]: The previous description has never defined what "scene root" was, so I don't see this as a breaking change. We've also agreed that `skeleton` doesn't affect node transforms. The updated readme says that the common root (node) must exist.

Comment (donmccurdy) [reply specification/2.0/schema/skin.schema.json]: It seems technically possible that assets may exist that do not have a common root node, although I don't know of any authoring tools that would create them that way, or could ingest them that way.

@bghgary assuming this seems OK to you as well i'm fine to make the spec more strict.

Comment (bghgary) [reply specification/2.0/schema/skin.schema.json]: Seems okay to me.

Comment (javagl) [reply specification/2.0/schema/skin.schema.json]: I'd have to review some of the discussion around the "skeleton" (particularly, the extensive information in 1270). But doesn't the statement

> "The node must be the closest common root of the joints hierarchy or a direct or indirect parent node of the closest common root."

boil down to something like "The node must be a common ancestor of all joints"? When it can be a direct or indirect *parent* of the closest common root, then the part of the statement that it must be the *closest common root* does not seem to make sense. 

(Sorry, maybe I overlooked or misunderstood something here...)

Comment (lexaknyazev) [reply specification/2.0/schema/skin.schema.json]: @donmccurdy @bghgary 
Any opinions on wording here wrt @javagl comment? I'm fine either way.

Comment (bghgary) [reply specification/2.0/schema/skin.schema.json]: I think both versions read okay to me. @lexaknyazev version calls out more explicitly that it doesn't have the be the _closest_ common root. I'm reading the statement as "The node be [A] or [B]" so they don't conflict with each other.

Comment (javagl) [reply specification/2.0/schema/skin.schema.json]: As mentioned in the comment below: I think that both wordings technically mean the same thing, so it may be just a matter of subjective preference.

**Comment by pjcozzi** [comment]
@bghgary @donmccurdy OK to merge this?

**Comment by lexaknyazev** [comment]
@donmccurdy 
"Property Reference" section should have been updated before merging.

Also, I think that we could streamline language a bit. The updated spec says:

> When a skin is referenced by a node within a scene, all joints used by the skin must belong to the same scene. Joint nodes must have a common root that may or may not be a joint node itself.

Since a common root is required, a node can have only one parent, and scenes can refer to only root-level nodes, we can rewrite this part like:

> Joint nodes must have a common root that may or may not be a joint node itself. When a skin is referenced by a node within a scene, the common root must belong to the same scene.

**Comment by donmccurdy** [comment]
I'd rephrase the first sentence slightly:

> Each skin's joints must have a common root, which may or may not be a joint node itself. When a skin is referenced by a node within a scene, the common root must belong to the same scene.

**Comment by lexaknyazev** [comment]
Opened #1573.
